packages/gatsby-plugin-netlify/src/create-redirects.js

```diff
@@ -1,16 +1,25 @@
-import fs from "fs"
-import pify from "pify"
+import { appendFile, exists, writeFile } from "fs-extra"
 
-const writeFile = pify(fs.writeFile)
-
-export default function writeRedirectsFile(pluginData, redirects) {
+export default async function writeRedirectsFile(pluginData, redirects) {
   const { publicFolder } = pluginData
 
-  // https://www.netlify.com/docs/redirects/
-  const data = redirects.map(redirect => {
-    const status = redirect.isPermanent ? 301 : 302
-    return `${redirect.fromPath}  ${redirect.toPath}  ${status}`
-  })
+  if (redirects.length > 0) {
+    const FILE_PATH = publicFolder(`_redirects`)
+
+    // https://www.netlify.com/docs/redirects/
+    redirects = redirects.map(redirect => {
+      const status = redirect.isPermanent ? 301 : 302
+      return `${redirect.fromPath}  ${redirect.toPath}  ${status}`
+    })
+
+    const data = `## Created with gatsby-plugin-netlify\n\n${redirects.join(
+      `\n`
+    )}`
+
+    const fileExists = await exists(FILE_PATH)
 
-  return writeFile(publicFolder(`_redirects`), data.join(`\n`))
+    return fileExists
```

Comment:
The problem with this is that files in public aren't deleted between builds so each time you build the redirects file will just grow and grow.

It seems like a fairly rare thing that someone would both want to maintain a handwritten redirects file & have this plugin generating one. Perhaps make it a plugin option? E.g. `appendAutogeneratedRedirectsToStaticRedirectsFile` haha or a shorter option name :-D

Reply:
Oh interesting. I didn't realize that b'c I've been manually clearing between builds.

Reply:
My understanding is this: If a file exists in `static/_redirects` it will be copied over each time the build command is run so it's safe to append. If no file exists, we'll create one initially and then (unless the user manually clears the contents of `public`) that same file will be around the next time the build command is run.

If that's the case, we can just search for our "_Created with gatsby-plugin-netlify_" comment in order to determine if it's the file we created or not. It's a little hacky I guess, but I think it should work like:

* Is there a file?
  * **No**: Write a new one.
  * **Yes**: Does it contain the "_Created with gatsby-plugin-netlify_" comment?
     * **No**: Then it's a user-generated file, copied from `static`; append to it.
     * **Yes**: Then it's a file we created previous; override it.

I'll push an update with this proposed change momentarily. I've tested it locally and it seems to work correctly.

Reply:
Love it! Complicated implementation > complicated UX. Seems like it should be pretty stable.